drivers/include/sx127x.h

```diff
@@ -0,0 +1,704 @@
+ * @brief   SX127X initialization result.
+ */
+enum {
+    SX127X_INIT_OK = 0,                /**< Initialization was successful */
+    SX127X_ERR_SPI,                    /**< Failed to initialize SPI bus or CS line */
+    SX127X_ERR_TEST_FAILED,            /**< SX127X testing failed during initialization (check chip) */
+    SX127X_ERR_THREAD                  /**< Unable to create DIO handling thread (check amount of free memory) */
+};
+
+/**
+ * @brief   Radio driver supported modems.
+ */
+enum {
+    SX127X_MODEM_FSK = 0,              /**< FSK modem driver */
+    SX127X_MODEM_LORA,                 /**< LoRa modem driver */
+};
+
+/**
+ * @brief   LoRa signal bandwidth.
+ */
+enum {
+    SX127X_BW_125_KHZ = 0,             /**< 125 kHz bandwidth */
+    SX127X_BW_250_KHZ,                 /**< 250 kHz bandwidth */
+    SX127X_BW_500_KHZ                  /**< 500 kHz bandwidth */
+};
+
+/**
+ * @brief   LoRa spreading factor rate
+ */
+enum {
+    SX127X_SF6 = 6,                    /**< spreading factor 6 */
+    SX127X_SF7,                        /**< spreading factor 7 */
+    SX127X_SF8,                        /**< spreading factor 8 */
+    SX127X_SF9,                        /**< spreading factor 9 */
+    SX127X_SF10,                       /**< spreading factor 10 */
+    SX127X_SF11,                       /**< spreading factor 11 */
+    SX127X_SF12                        /**< spreading factor 12 */
+};
+
+/**
+ * @brief   LoRa error coding rate.
+ */
+enum {
+    SX127X_CR_4_5 = 1,                 /**< coding rate 4/5 */
+    SX127X_CR_4_6,                     /**< coding rate 4/6 */
+    SX127X_CR_4_7,                     /**< coding rate 4/7 */
+    SX127X_CR_4_8                      /**< coding rate 4/8 */
+};
+
+/**
+ * @brief   Radio driver internal state machine states definition.
+ */
+enum {
+    SX127X_RF_IDLE = 0,                /**< Idle state */
+    SX127X_RF_RX_RUNNING,              /**< Sending state */
+    SX127X_RF_TX_RUNNING,              /**< Receiving state */
+    SX127X_RF_CAD,                     /**< Channel activity detection state */
+};
+
+/**
+ * @brief   Event types.
+ */
+enum {
+    SX127X_RX_DONE = 0,                /**< Receiving complete */
+    SX127X_TX_DONE,                    /**< Sending complete*/
+    SX127X_RX_TIMEOUT,                 /**< Receiving timeout */
+    SX127X_TX_TIMEOUT,                 /**< Sending timeout */
+    SX127X_RX_ERROR_CRC,               /**< Receiving CRC error */
+    SX127X_FHSS_CHANGE_CHANNEL,        /**< Channel change */
+    SX127X_CAD_DONE,                   /**< Channel activity detection complete */
+};
+
+/**
+ * @brief   LoRa configuration structure.
+ */
+typedef struct {
+    uint8_t power;                     /**< Signal power */
+    uint8_t bandwidth;                 /**< Signal bandwidth */
+    uint8_t datarate;                  /**< Spreading factor rate, e.g datarate */
+    bool low_datarate_optimize;        /**< Optimize for low data rate */
```

Comment:
What about these bools? Also this and the other structs could benefit of some [reordering](http://www.catb.org/esr/structure-packing/)